packages/grid/_modules_/grid/components/GridFooter.tsx

```diff
@@ -31,21 +29,17 @@ export const GridFooter = React.forwardRef<HTMLDivElement, GridFooterContainerPr
         <GridRowCount rowCount={totalRowCount} visibleRowCount={visibleRowCount} />
       ) : null;
 
-    const PaginationComponent =
-      !!rootProps.pagination &&
-      paginationState.pageSize != null &&
+    const paginationElement = rootProps.pagination &&
       !rootProps.hideFooterPagination &&
-      apiRef?.current.components.Pagination;
-
-    const PaginationElement = PaginationComponent && (
-      <PaginationComponent {...apiRef?.current.componentsProps?.pagination} />
-    );
+      rootProps.components.Pagination && (
+        <rootProps.components.Pagination {...rootProps.componentsProps?.pagination} />
+      );
 
     return (
       <GridFooterContainer ref={ref} {...props}>
         {SelectedRowCountElement}
         {RowCountElement}
-        {PaginationElement}
+        {paginationElement}
```

Comment:
Great call, elements are camelCase, not PascalCase.

Reply:
Their are some others to rename, I'll to a quick standalone PR later